app/script/components/messageTimerButton.js

```diff
@@ -0,0 +1,91 @@
+/*
+ * Wire
+ * Copyright (C) 2018 Wire Swiss GmbH
+ *
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License as published by
+ * the Free Software Foundation, either version 3 of the License, or
+ * (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program. If not, see http://www.gnu.org/licenses/.
+ *
+ */
+
+'use strict';
+
+window.z = window.z || {};
+window.z.components = z.components || {};
+
+z.components.MessageTimerButton = class MessageTimerButton {
+  constructor(params) {
+    this.conversationEntity = params.conversation;
+    this.hasMessageTimer = ko.pureComputed(() => {
+      return this.conversationEntity() ? this.conversationEntity().messageTimer() : false;
+    });
+    this.isTimerDisabled = ko.pureComputed(() => this.conversationEntity().hasGlobalMessageTimer());
+    this.timerText = ko.pureComputed(() => {
+      if (this.hasMessageTimer()) {
+        return z.util.TimeUtil.formatDuration(this.conversationEntity().messageTimer());
+      }
+      return {};
```

Comment:
Isn't it supposed to be a text. I'd go for a `return '';` in order to be consistent (and for an easy potential typescript migration :heart: )